pkg/cloudprovider/cloudprovider.go

```diff
@@ -340,20 +291,55 @@ func (c *CloudProvider) resolveInstanceTypes(ctx context.Context, machine *v1alp
 	return lo.Filter(instanceTypes, func(i *cloudprovider.InstanceType, _ int) bool {
 		return reqs.Compatible(i.Requirements) == nil &&
 			len(i.Offerings.Requirements(reqs).Available()) > 0 &&
-			resources.Fits(resources.Merge(machine.Spec.Resources.Requests, i.Overhead.Total()), i.Capacity)
+			resources.Fits(machine.Spec.Resources.Requests, i.Allocatable())
 	}), nil
 }
 
-// TODO @joinnis: Remove name resolution when Machine migration is completed
-func (c *CloudProvider) instanceToMachine(ctx context.Context, instance *ec2.Instance, instanceTypes []*cloudprovider.InstanceType) (*v1alpha5.Machine, error) {
-	machine := &v1alpha5.Machine{}
-	instanceType, found := lo.Find(instanceTypes, func(i *cloudprovider.InstanceType) bool {
-		return aws.StringValue(instance.InstanceType) == i.Name
+func (c *CloudProvider) resolveInstanceTypesFromInstance(ctx context.Context, instance *ec2.Instance) ([]*cloudprovider.InstanceType, error) {
```

Comment:
this is a weird function because it's not obvious how it's resolving instance types from an instance. I would expect, based on the name, to return a single instance type. Is it even worth wrapping this functionality, rather than just resolving the provisioner and then getting instance types from that in-line?